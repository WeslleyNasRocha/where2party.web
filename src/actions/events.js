import { toast } from 'react-toastify';
import database from '../firebase/firebase';

export const addEvent = event => ({
  type: 'ADD_EVENT',
  event,
});

export const startAddEvent = (eventData = {}) => (dispatch) => {
  // const {
  //   descricao, titulo, local, data, image, orgId, address,
  // } = eventData;
  // const event = {
  //   descricao,
  //   titulo,
  //   local,
  //   data,
  //   image,
  //   orgId,
  //   address,
  // };
  const {
    Titulo, Descricao, Address, Data, Local,
  } = eventData;
  const event = {
    Titulo,
    Descricao,
    Address,
    Data,
    Local,
  };
  console.log(eventData);
  database
    .ref('eventos')
    .push(event)
    .then((ref) => {
      toast.success('Evento Adicionado com sucesso');
      dispatch(addEvent({
        id: ref.key,
        ...event,
      }));
    });
};

export const removeEvent = ({ id } = {}) => ({
  type: 'REMOVE_EVENT',
  id,
});

export const editEvent = (id, updates) => ({
  type: 'EDIT_EVENT',
  id,
  updates,
});

export const setEvents = events => ({
  type: 'SET_EVENTS',
  events,
});

export const startSetEvent = () => dispatch =>
  database
    .ref('eventos')
    .once('value')
    .then((snapshot) => {
      const events = [];

      snapshot.forEach((childSnapshot) => {
        events.push({
          id: childSnapshot.key,
          ...childSnapshot.val(),
        });
      });

      dispatch(setEvents(events));
    });
