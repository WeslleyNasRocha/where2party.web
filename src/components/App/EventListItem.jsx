import React from 'react';
import { Link } from 'react-router-dom';

const EventListItem = ({ id, description, titulo }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{titulo}</h3>
    </Link>
    <p>{description}</p>
  </div>
);

export default EventListItem;
