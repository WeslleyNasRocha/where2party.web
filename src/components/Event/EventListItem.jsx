import React from 'react';
import { Link } from 'react-router-dom';

export const EventListItem = ({
  id, Titulo, Descricao, Address, Data, Local,
}) => (
  <div>
    <div className="box box-widget widget-user">
      {/* <Link to={`/edit/${id}`}> */}
      <Link to={`/edit/${id}`}>
        <h3>{Titulo}</h3>
      </Link>
      {/* </Link> */}
      <p>Descricao: {Descricao}</p>
      <p>Data: {Data}</p>
      <p>Address: {Address}</p>
      <p>
        Local: <b>long: </b>
        {Local.longitude} <b>lat: </b>
        {Local.latitude}
      </p>
    </div>
  </div>
);

export default EventListItem;
