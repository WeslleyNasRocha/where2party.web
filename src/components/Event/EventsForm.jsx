import React, { Component } from 'react';
import { Formik, Field, Form } from 'formik';
import moment from 'moment';
import DatePicker from 'react-datepicker';
import { toast } from 'react-toastify';
import PlacesAutocomplete, { geocodeByAddress, getLatLng } from 'react-places-autocomplete';

import 'react-datepicker/dist/react-datepicker.css';

import Yup from 'yup';

export const EventsForm = props => (
  <div>
    <div className="box-body">
      <Formik
        initialValues={{
          Titulo: '',
          Descricao: '',
          Address: '',
          Data: moment().format('DD/MM/YY'),
          Local: {},
        }}
        onSubmit={(values) => {
          let submitValues = values;

          geocodeByAddress(values.Address)
            .then(results => getLatLng(results[0]))
            .then((latLng) => {
              submitValues = {
                ...submitValues,
                Local: {
                  latitude: latLng.lat,
                  longitude: latLng.lng,
                  latitudeDelta: 0.015,
                  longitudeDelta: 0.012,
                },
              };
              // console.log(submitValues);
              props.submit(submitValues);
            })
            .catch(error => toast.error(error));
        }}
        // validationSchema={Yup.object().shape({
        //   titulo: Yup.string()
        //     .min(5)
        //     .required(),
        //   description: Yup.string()
        //     .min(10)
        //     .max(250),
        // })}
        render={({
 touched, errors, values, setFieldValue, setTouched,
}) => (
  <Form>
    <div className="form-group">
      <label htmlFor="Titulo" className="col-sm-2 control-label">
                Titulo
      </label>

      <div className="col-sm-10">
        <Field
          type="Text"
          name="Titulo"
          className="form-control"
          id="Titulo"
          placeholder="Titulo"
        />
      </div>
      {touched.Titulo &&
                errors.Titulo && (
                  <div style={{ marginTop: 10 }} className="alert alert-danger" role="alert">
                    <p>{errors.Titulo}</p>
                  </div>
                )}
    </div>
    <div className="form-group">
      <label htmlFor="Descricao" className="col-sm-2 control-label">
                Descrição
      </label>
      <div className="col-sm-10">
        <Field
          name="Descricao"
          id="Descricao"
          type="text"
          className="form-control"
          placeholder="Descrição"
        />
      </div>
      {touched.Descricao &&
                errors.Descricao && (
                  <div style={{ marginTop: 10 }} className="alert alert-danger" role="alert">
                    <p>{errors.Descricao}</p>
                  </div>
                )}
    </div>
    <div className="form-group">
      <label htmlFor="Data" className="col-sm-2 control-label">
                Data
      </label>
      <div className="col-sm-10">
        <DatePicker
          name="Data"
          dateFormat="DD/MM/YYYY"
          value={values.Data}
          onChange={(value) => {
                    setTouched('Data', true);
                    setFieldValue('Data', value.format('DD/MM/YY'));
                  }}
        />
      </div>
    </div>
    <div className="form-group">
      <label htmlFor="Address" className="col-sm-2 control-label">
                Endereço
      </label>
      <div className="col-sm-10">
        <PlacesAutocomplete
          inputProps={{
                    value: values.Address,
                    onChange: address => setFieldValue('Address', address),
                  }}
          style={{
                    autocompleteContainer: {
                      position: 'absolute',
                      top: '100%',
                      backgroundColor: 'white',
                      border: '1px solid #555555',
                      width: '100%',
                    },
                    autocompleteItem: {
                      backgroundColor: '#ffffff',
                      padding: '10px',
                      color: '#555555',
                      cursor: 'pointer',
                    },
                    autocompleteItemActive: {
                      backgroundColor: '#fafafa',
                    },
                    googleLogoContainer: {
                      textAlign: 'right',
                      padding: '1px',
                      backgroundColor: '#fafafa',
                    },
                    googleLogoImage: {
                      width: 150,
                    },
                  }}
        />
      </div>
      {touched.Address &&
                errors.Address && (
                  <div style={{ marginTop: 10 }} className="alert alert-danger" role="alert">
                    <p>{errors.Address}</p>
                  </div>
                )}
    </div>

    <button type="submit" className="btn btn-info btn-block">
              save
    </button>
  </Form>
        )}
      />
    </div>
  </div>
);

export default EventsForm;
