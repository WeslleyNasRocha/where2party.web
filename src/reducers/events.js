const eventReducerDefaultState = [];

const eventReducer = (state = eventReducerDefaultState, action) => {
  // console.log(action.type, action);
  switch (action.type) {
    case 'ADD_EVENT':
      return [...state, action.event];
    case 'REMOVE_EVENT':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_EVENT':
      return state.map((event) => {
        if (event.id === action.id) {
          return {
            ...event,
            ...action.updates,
          };
        }
        return event;
      });
    case 'SET_EVENTS':
      return action.events;
    // return ['carai'];
    default:
      return state;
  }
};

export default eventReducer;
