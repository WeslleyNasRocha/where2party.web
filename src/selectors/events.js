import moment from 'moment';

// Get visible events

export default (events, {
  text, startDate, endDate,
}) =>
  events
    .filter((event) => {
      const dataEvent = moment(event.data, 'DD/MM/YY');
      const createdAtMoment = moment(dataEvent);
      const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
      const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
      const textMatch = event.titulo.toLowerCase().includes(text.toLowerCase());

      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => (a.createdAt < b.createdAt ? 1 : -1));
