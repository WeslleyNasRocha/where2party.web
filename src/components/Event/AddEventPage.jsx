import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import EventsForm from './EventsForm';
import { startAddEvent, addEvent } from '../../actions/events';

export class AddEventPage extends Component {
  onSubmit = (event) => {
    this.props.startAddEvent(event);
    // .then(() => {
    //   // this.props.history.push('/events');
    //   console.log('foi');
    // })
    // .catch(e => console.log(e));
  };
  render() {
    return (
      <div className="col-md-6">
        <div className="box box-success">
          <div className="box-header with-border">
            <h2 className="box-title">Add Event Page</h2>
          </div>
          <EventsForm submit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddEvent: event => dispatch(startAddEvent(event)),
});

export default connect(undefined, mapDispatchToProps)(AddEventPage);
