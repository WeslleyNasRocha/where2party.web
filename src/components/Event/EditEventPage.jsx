import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import EventsForm from './EventsForm';

import { startEditEvent, startRemoveEvent } from '../../actions/events';

export class EditEventPage extends Component {
  onSubmit = (event) => {
    // console.log(event);
    this.props.startEditEvent(event, this.props.evento.id);
    this.props.history.push('/events');
  };
  onRemove = () => {
    this.props.startRemoveEvent(this.props.evento);
    this.props.history.push('/events');
  };
  render() {
    return (
      <div className="col-md-10">
        <div className="box box-warning">
          <div className="box-header with-border">
            <h2 className="box-title">Editar Evento</h2>
          </div>
          <EventsForm submit={this.onSubmit} evento={this.props.evento} remove={this.onRemove} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startEditEvent: (updates, id) => dispatch(startEditEvent(updates, id)),
  startRemoveEvent: ({ id }) => dispatch(startRemoveEvent({ id })),
});

const mapStateToProps = (state, props) => ({
  evento: state.events.find(evento => evento.id === props.match.params.id),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditEventPage);
