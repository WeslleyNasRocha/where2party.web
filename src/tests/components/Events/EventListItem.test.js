import React from 'react';
import { shallow } from 'enzyme';
import { EventListItem } from '../../../components/Event/EventListItem';
import events from '../../fixtures/events';

test('Should render eventlistitem correctly', () => {
  const wrapper = shallow(<EventListItem {...events[0]} />);
  expect(wrapper).toMatchSnapshot();
});
