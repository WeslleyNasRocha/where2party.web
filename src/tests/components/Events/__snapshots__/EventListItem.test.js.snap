// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render eventlistitem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Chacrinha
    </h3>
  </Link>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vulputate mi in turpis dapibus facilisis.
  </p>
  <p />
  <p>
    10/01/17
  </p>
  <p>
    X2e8wZADnwhU3EonNmH6ppqO9g73++1505237223764.jpeg
  </p>
  <p>
    X2e8wZADnwhU3EonNmH6ppqO9g73
  </p>
  <p />
</div>
`;
